### ISSUE TITLE
Fix: added new dng sdk 1.5.x

### ISSUE THREAD

**Issue by Jamaika1** [gopro/gpr#30]
I don't know will library have update?
I am adding patches to the latest dng sdk 1.5.x. Maybe it will be useful to someone.
https://github.com/Jamaika1/dng_sdk_1.5.1_-_gpr_sdk_1.0.0

**Comment by OgreTransporter** [comment]
Wonderful! Thank you very much! Why not create a PR to update the gpr library? I have already started updating expat, zlib, xmp, and cJSON. I have also included your changes to dng and gpr. But you obviously also changed VC5. For example VC5_ENCODER_PIXEL_FORMAT_RGGB_16 (https://github.com/Jamaika1/dng_sdk_1.5.1_-_gpr_sdk_1.0.0/blob/f9a54f80c5bd8eefa8f906824198d1b1628febeb/gpr_sdk/gpr.cpp#L157) is not included in the official code.
```
    typedef enum
    {
        VC5_ENCODER_PIXEL_FORMAT_RGGB_12 = 0,           // RGGB 14bit pixels packed into 16bits

        VC5_ENCODER_PIXEL_FORMAT_RGGB_12P = 1,           // RGGB 14bit pixels packed into 16bits

        VC5_ENCODER_PIXEL_FORMAT_RGGB_14 = 2,           // RGGB 14bit pixels packed into 16bits
        
        VC5_ENCODER_PIXEL_FORMAT_GBRG_12 = 3,           // GBRG 12bit pixels packed into 16bits
        
        VC5_ENCODER_PIXEL_FORMAT_GBRG_12P = 4,          // GBRG 12bit pixels packed into 12bits
        
        VC5_ENCODER_PIXEL_FORMAT_DEFAULT = VC5_ENCODER_PIXEL_FORMAT_RGGB_14,
        
    } VC5_ENCODER_PIXEL_FORMAT;
```
I guess you took the codec from the CineForm SDK and then rewrote the functions vc5_encoder_process (vc5_encoder.c) and vc5_decoder_process (vc5_decoder.c)? I am stuck at that point.

**Comment by Jamaika1** [comment]
I shouldn't admit, but I'm not a programmer. As a programming hobbyist, I wanted to run the libraw library from DNG SDK and GPR SDK. The creator of libraw added a function raw 8/12/14/16bit. 
```
#ifdef USE_6BY9RPI
	void		rpi_load_raw8();
	void		rpi_load_raw12();
	void		rpi_load_raw14();
	void		rpi_load_raw16();
	void		parse_raspberrypi();
#endif
``` 
I also added GoPro GPR bayerphase 8/12/14/16bit. He realizes that it isn't connected.
https://github.com/gopro/gpr/tree/bayerphase
What is the libraw problem. The converter has a bad colormatrix for DNG SDK, but dng_valitable seems to be valid.

**Comment by OgreTransporter** [comment]
> This may be due to the imperfections of the gcc 10/11 software.

Not necessarily. The code is partly written very sloppily and contains numerous errors.

Example:
https://github.com/gopro/gpr/blob/e956c5961b5f046a6fda1b2f428b7718e9008122/source/app/common/argument_parser/argument_parser.cpp#L88

The function `get_application_path` returns the variable `application_path`:
https://github.com/gopro/gpr/blob/e956c5961b5f046a6fda1b2f428b7718e9008122/source/app/common/argument_parser/argument_parser.h#L39

However, this variable is not set anywhere, which means that there are memory access errors. Actually all member variables of a class should be set in the constructor to avoid such errors. Alternatively you can use STL data types. Bugfix: https://github.com/OgreTransporter/gpr/commit/10d7e9c77854d62cd0e3c61fcf601f1ebf752e19

I used your code and did a complete upgrade of GPR. Thanks again for that! Probably there are still some minor corrections to be made.

https://github.com/OgreTransporter/gpr
